pkg/services/alerting/rule.go

```diff
@@ -111,11 +113,33 @@ func NewRuleFromDBAlert(ruleDef *m.Alert) (*Rule, error) {
 
 	for _, v := range ruleDef.Settings.Get("notifications").MustArray() {
 		jsonModel := simplejson.NewFromAny(v)
-		id, err := jsonModel.Get("id").Int64()
-		if err != nil {
-			return nil, ValidationError{Reason: "Invalid notification schema", DashboardId: model.DashboardId, Alertid: model.Id, PanelId: model.PanelId}
+		if id, err := jsonModel.Get("id").Int64(); err == nil {
+			cmd := m.GetAlertNotificationsQuery{
```

Comment:
@bergquist I "convert" id to uid here using sql request. I thought, if id automatically migrates in db, so I can use migrated uid.

Reply:
Awesome! Happy new years! :)

Reply:
@bergquist I fixed. But I've got a problem:
1. I create notification channel and notification has got random uid and for example id 5.
2. I add new notification channel to alert rules
3. Alert settings contain id=5 for this channel.
4. Function NewAlertFromDb notification with uid=000000005

Reply:
Not sure I understand what the problem is. 

When we add new notification channels in this PR we should always add the UID. We should only support `ID` for alerts that already exist in the database until they are saved again next time.

Reply:
So this (https://github.com/grafana/grafana/blob/master/pkg/api/dtos/alerting.go#L66) struct needs the new `uid` property so it can be sent to the frontend.

This is starting to look really good! Great work :)

Reply:
Yes, old id will be converted properly with fmt and DB migration. But new notifications won't work in all cases.
@bergquist 
Alert after PR uses uid and has 3 cases:
1. Old notification ids. Ids are converted to uid via fmt(your proposal) and migration do the same. They are matched. It works!
2. New notification channels, that were created by custom config with uid. It works!
3. New notification channel which were created in browser. Ids of these channels are added to settings as well as for old notifications on frontend. But backend generated random uid in creation moment.Generated uid is not matched with id from fmt. It doesn't work!
 func NewRuleFromDBAlert at first checks id and converts id to uid vai fmt(for old and new notifications from browser). When id doesn't exist, checks uid.
That's it))

You've proposed to use fmt for id to uid conversion, I've proposed to get uid by id from DB for 1st and 3rd cases.

Yes, it's possible to add uid https://github.com/grafana/grafana/blob/master/pkg/api/dtos/alerting.go#L66 , but without updating this https://github.com/grafana/grafana/blob/master/public/app/features/alerting/AlertTabCtrl.ts#L145 it won't work. It will add to me additionally work to update frontend and frontend tests. It could be next step to convert all request from frontend to backed to use uid.
But in the beginning you wanted to change only backend part) And I agreed with it)

This commit https://github.com/grafana/grafana/pull/14229/commits/553a601a4d8fbe40d99e9d49c1bbad6b4381891e with getting uid by id from db.

Current commit is using fmt.